### DISCUSSION TITLE
Suggested simple enhancement to magic tags

### DISCUSSION THREAD

**Discussion by gwhitney** [pods-framework/pods#6761] [Ideas]
I find the following filter to be extremely useful:

```php
add_filter('pods_pods_field_text', 'pods_extend_magic_field_attributes', 10, 5);
function pods_extend_magic_field_attributes($val, $fieldinfo, $row, $params, $pod) {
	if (count($params->traverse) == 2 && isset($fieldinfo[$params->traverse[1]])) {
		return $fieldinfo[$params->traverse[1]];
	}
	return $val;
}
```
It makes attributes of a field such as its label accessible via shortcode, as in `{@myfield.label}`.  I had expected that such magic tags would work out of the box; they seem to mesh with other uses of "." in magic tags. So I just wanted to suggest that you might want to incorporate this behavior into the core of Pods, or if not, let me know where it would be most useful to post this example so that others might benefit from it.  Thanks,

**Comment by quasel** [comment]
might this be an idea @sc0ttkclark ? or is that more an item for the docs @jimtrue ;)

**Comment by jimtrue** [comment]
I'd have to see how it interacted with traversal myself. I think it's a great idea, but it could be confusing since that's the same syntax we use for Traversal of relationships, etc.

**Reply by gwhitney** [reply]
Right, which is why I was expecting it to work with Pods "out of the box"... I still use this notation with my tweak in place, and I don't find it's confusing since it's how I expected things to work in the first place. But others' mileage may vary.
